### PR TITLE
[IMPROVEMENT] AQC111U Driver v1.3.3.0

### PR DESCRIPTION
**Pull request by itnok**
Thanks @stormi for providing the dummy LFS file to allow the push in my fork.

Hopefully, this PR satisfies your standards, if not please feel free to point out what I need to correct to comply.

As you might have noticed I changed the name of the repo to aqc111u-module _(added the `u` part)_ this, might not fully adhere to your policy as the final kernel module produced it is still called `aqc111`, but it is IMHO a better naming strategy considering that an `aqc111` driver does exist and, despite addressing the same chipset, does not work for NICs using USB3 at transport layer _(it's for the regular PCIe NICs)_. The `u` in the driver name stands for USB and is the only differentiating aspect between `aqc111` and `aqc111u`.

Two possible approaches if you do not like having the name of the repo changed while the produced module is still `aqc111`:

- Altering the code provided by Marvell _(is GPL and as long as all authors are mentioned is fine I believe)_ to produce a `aqc111u` kmodule instead;
- Keep the name as `aqc111-module` for the repo _(this driver is already in the Linux Kernel upstream since 5.12-rc6 supporting the `aqc111u` not to be confused with the Aquantia Atlantic which supports `aqc111`, `aqc200` and others)_

Poor naming convention choices if you ask me... but nevertheless that's what we currently have.

Other than that, you will see a much _"puffier"_ README.md _(not sure it that's allowed)_ and the fact that instead of pulling the drivers directly from the website _(of Sabrent originally or Marvell, even better)_ the source code has been added to the repo directly, hence the LFS problem)_

Thanks

### CODE REVIEW

File: README.md
Comment (stormi) [inline README.md]: Those are generic instructions for building packages that would apply to all of our RPMs. In addition, they only address the use case of people wanting to build things by themselves without installing from the XCP-ng repositories. The process used to build the packages for the repositories is different and uses Koji.

In the future, it would be a useful enhancement to have our READMEs link to an appropriate "how to build locally" doc in https://xcp-ng.org/docs/, but it's not ready yet.

So let's remove this section from this readme for now.

File: README.md
Comment (stormi) [inline README.md]: This should probably be a PR to the xcp-ng-build-env repository instead, to the dockerfiles themselves, if that's going to work for everybody. This, plus an update of the README where it mentions the potentiel issue with user ids.

File: README.md
Comment (stormi) [inline README.md]: As those are instructions for users, they rather belong to https://xcp-ng.org/docs/. Probably somewhere under https://xcp-ng.org/docs/hardware.html#unlisted-hardware. It would also be good to split it in two sections:
* one dedicated to this module, if needed (for generic information about it and pointing at the name of the package that people will need to install from the xcp-ng repos)
* one dedicated to managing USB NICs with udev. We've had various forum threads or github issues about this requesting for us to support this kind of configuration. After reviewing them, if your solution allows to workaround the issues for them too, this would be a useful addition to our docs.

File: README.md
Comment (stormi) [inline README.md]: This probably won't age well, unless you maintain it. And I think it would be better placed in https://xcp-ng.org/docs/hardware.html#unlisted-hardware

**Review by stormi — COMMENTED** [review]
Thanks for the PR.

About the naming (the situation is indeed confusing), I agree that `aqc111u` is a better name, and I think `aqc111u.ko` would be a better filename too, in order not to clash with upstream kernel modules in the future. The better way to fix, besides fixing it upstream if possible, is through a patch applied during the `%prep` phase (Note: changing `%setup` to `%autosetup` will apply any patch declared in the spec file.

**Comment by stormi** [comment]
Please also amend your commit message to add a DCO (it's a legal thing that means you certify that the code you are contributing is either yours or something that you are allowed to contribute, which is the case here).

Reference: https://wiki.linuxfoundation.org/dco

~~We'll add automated checks for that kind of thing in the future to PRs, but it's not ready yet.~~ Update: turns out enforcing it for a whole github organization was just a few clicks. Done now.

**Comment by stormi** [comment]
I just renamed the repository to `aqc111u-module`

File: README.md
Comment (itnok) [reply README.md]: Addressed with [PR#14 in `xcp-ng-build-env`](https://github.com/xcp-ng/xcp-ng-build-env/pull/14), plus some _"extra goodies"_...

File: README.md
Comment (itnok) [reply README.md]: How do you prefer me to address this?

- the changes should be removed and the `https://xcp-ng.org/docs/` site should have the modifications added? _(How?)_
- leave the info here and just split it as suggested?

Thanks

File: README.md
Comment (itnok) [reply README.md]: The documentation, as you pointed out, is probably the right place, although until it's missing there it is an interesting chunk of information to have. How do you prefer me to address this?

- the changes should be removed and the `https://xcp-ng.org/docs/` site should have the modifications added? (How?)
- leave the info here and maintain it? _(it will not age well, but it will not age that fast I believe)_

**Comment by itnok** [comment]
> Please also amend your commit message to add a DCO (it's a legal thing that means you certify that the code you are contributing is either yours or something that you are allowed to contribute, which is the case here).
> 
> Reference: https://wiki.linuxfoundation.org/dco
> 
> ~We'll add automated checks for that kind of thing in the future to PRs, but it's not ready yet.~ Update: turns out enforcing it for a whole github organization was just a few clicks. Done now.

Force pushed the same commit with added `Signed-Off`: is this enough to comply with the DCO _(if not please advise exactly how the change should look like)_

File: README.md
Comment (itnok) [reply README.md]: Addressed in 587c9e29

File: README.md
Comment (stormi) [reply README.md]: I would prefer to have it in the docs. Each page there has a "Help us improve this page" link that allows to make changes. Or you can directly create a PR against https://github.com/xcp-ng/xcp-ng-org/

File: README.md
Comment (stormi) [reply README.md]: I definitely prefer to have it in the docs :)

**Comment by stormi** [comment]
> Force pushed the same commit with added `Signed-Off`: is this enough to comply with the DCO _(if not please advise exactly how the change should look like)_

Yes, it's enough, thanks!

File: SPECS/aqc111u-module.spec
Comment (stormi) [inline SPECS/aqc111u-module.spec]: Since there has been no build between your commits, it all belongs to a single changelog entry. 

Other comments:
- It is customary to add a newline between changelog entries (here it would be between lines 64 and 65 if we were to keep both entries)
- The version number in the changelog must also include the release of the RPM, so here `1.3.3.0-1`.

**Comment by stormi** [comment]
So, sorry about the delay. I'm usually a lot faster to review PRs but I've been away for 2 weeks. Again many thanks for your contributions, and let's finalize this PR soon! Then I'll send a build to our build system and make the RPM available in our testing repositories.

FYI, once the RPM is made available, we'll also update https://github.com/xcp-ng/xcp/wiki/Drivers#other-kernel-modules-available-in-xcp-ng-82

**Comment by itnok** [comment]
My turn to apologize for the delay, our time off schedule collided in the worse possible way, I guess!

Will get back to this PR later today...

File: SPECS/aqc111u-module.spec
Comment (itnok) [reply SPECS/aqc111u-module.spec]: Hopefully, cd1afa2 fixes the non-standard formatting _(if not, please let me know)_

File: README.md
Comment (itnok) [reply README.md]: Commit 0f08735 removes this section in favor of a the [dedicated PR #97 in xcp-ng/xcp-ng-org documetation](https://github.com/xcp-ng/xcp-ng-org/pull/97)

File: README.md
Comment (itnok) [reply README.md]: Commit 0f08735 removes the section dedicated to the AQC111U information in favor of a the [dedicated PR #97 in xcp-ng/xcp-ng-org documetation](https://github.com/xcp-ng/xcp-ng-org/pull/97)

Commit d6b61b2 removes _(without any replacement elsewhere)_ the section dedicated to the use of `udev` rules to avoid pollution of unused PIFs upon reboot _(or pugging/unplugging)_ that applies, in general, to all USB NICs in XCP-ng. Not really sure where to put that content, but here it is clearly not the right place...

File: SPECS/aqc111u-module.spec
Comment (stormi) [inline SPECS/aqc111u-module.spec]: I don't know if there's a written rule about that, but in most distros the date number is put on two digits for better alignement. So `9` would become `09` here.

File: SOURCES/aqc111u-1.3.3.0.patch
Comment (stormi) [inline SOURCES/aqc111u-1.3.3.0.patch:79]: It did not occur to me during first review, but it will be tedious for you to update these file renaming patches over time (each time anything changes in the files), where a simple `mv` in the spec file would do the trick.

File: SPECS/aqc111u-module.spec
Comment (stormi) [inline SPECS/aqc111u-module.spec]: TIP: you might prefer to use %autosetup (that automatically applies patches) in case you have more patches in the future.

**Comment by stormi** [comment]
After the last changes (some of which are your call to make or not), please squash your commits before the merge.

File: SPECS/aqc111u-module.spec
Comment (itnok) [reply SPECS/aqc111u-module.spec]: Addressed in cd1afa2

File: SPECS/aqc111u-module.spec
Comment (itnok) [reply SPECS/aqc111u-module.spec]: Thanks for the tip! Addressed in e4b1caa.

File: SOURCES/aqc111u-1.3.3.0.patch
Comment (itnok) [reply SOURCES/aqc111u-1.3.3.0.patch:79]: Am not sure am following here, but I'd rather avoid logic in the spec file and have a patch to take care of as much as possible.

**Comment by itnok** [comment]
Please @stormi give it a final look if you do not mind... and if everything looks acceptable, please give me green light to merge _(will squash commits)_  🙏

File: SOURCES/aqc111u-1.3.3.0.patch
Comment (stormi) [reply SOURCES/aqc111u-1.3.3.0.patch:79]: As you wish, you will just have more work for each update :)

**Comment by stormi** [comment]
Green is the light !

**Comment by itnok** [comment]
Usually, the squash operation happens on merge, but hopefully what I did will be fine with you @stormi 

Should be finally RTM!

**Comment by stormi** [comment]
> Usually, the squash operation happens on merge, but hopefully what I did will be fine with you @stormi

Depends on the process :) I prefer to have the committer do the squash so that they can write the most appropriate commit message (but I do squash on merge sometimes, when the committer is not enough familiar with git).

File: SPECS/aqc111u-module.spec
Comment (stormi) [inline SPECS/aqc111u-module.spec]: The release number is still missing from the end of the line (`1.3.3.0` => `1.3.3.0-1`).

Comment (itnok) [reply SPECS/aqc111u-module.spec]: Must have misunderstood: thought there it was reported the version of the software packed without the packaging release. My bad! Thanks for triple-checking!!!

**Review by stormi — APPROVED** [review]

**Comment by stormi** [comment]
Congratulations, it's merged!

I also created the `8.2` branch that is the maintenance branch we'll start the koji builds from for XCP-ng 8.2. We'll keep `master` in sync with `8.2` until we start working on the next release, as we do for other packages.

Koji build: https://koji.xcp-ng.org//taskinfo?taskID=19311

If it succeeds, the RPM will land on the `xcp-ng-testing` RPM repository ~15 minutes afterwards.

Next steps: install it, test it, advertise it on the forum, and ideally have other people test it. Since it's an additional package without any impact on the rest of the system, we can move it to the updates repository after we got some feedback, to make it even easier to install, with a simple `yum install aqc111u-module` command.

[Update: changed the build link to a new one. The first one failed because I forgot to allow builds for that package on the V8.2 tag]

**Comment by stormi** [comment]
Please also update https://github.com/xcp-ng/xcp/wiki/Drivers#other-kernel-modules-available-in-xcp-ng-82 (and try to take notes somewhere to update it each time we release a newer version of the driver)

**Comment by itnok** [comment]
@stormi Should I edit the Wiki page directly or is there a process in place I should be aware of and follow _(also for approval?)_

**Comment by stormi** [comment]
You'll notice that I merged your commit without a merge commit. That's the way we do on our RPM source repositories in order to keep the git history as simple as possible.

**Comment by stormi** [comment]
> @stormi Should I edit the Wiki page directly or is there a process in place I should be aware of and follow _(also for approval?)_

You can edit the wiki directly :)

**Comment by itnok** [comment]
@stormi updated the Wiki in the section [Other kernel modules available in XCP-ng 8.2](https://github.com/xcp-ng/xcp/wiki/Drivers#other-kernel-modules-available-in-xcp-ng-82). It looks like the drivers were listed in alphabetical order therefore I put it at the top _(hope I guessed right...)_

**Comment by stormi** [comment]
The only thing I would have done differently is I wouldn't have put the release number after the version, this time :D.

**Comment by itnok** [comment]
Ooops... fixed! 🙏

**Pull request merged**
